src/tsickle_compiler_host.ts

```diff
@@ -29,9 +29,16 @@ export interface TsickleEnvironment {
    *  and generates a googmodule module name
    */
   pathToModuleName: (context: string, importPath: string) => string;
-  /** Tsickle treats warnings as errors, if true, ignore warnings */
+  /**
+   * Tsickle treats warnings as errors, if true, ignore warnings.  This might be
+   * useful for e.g. third party code.
+   */
   shouldIgnoreWarningsForPath: (filePath: string) => boolean;
-  /** Determines what we monkey patch the module.id to be in googmodule mode */
+  /**
+   * If we do googmodule processing, we monkey patch module.id, since that's
```

Comment:
Oh one last comment: I'm not sure "monkey patch" is the right term here -- that usually refers to patching out some API, but in this case we're backporting a newer API.  Maybe the term is "polyfill"?